TestApi/TestApi.DAL/Commands/AllocationService.cs

```diff
@@ -119,28 +131,21 @@ private async Task<Allocation> GetAllocationByUserId(Guid userId)
 
         private async Task CreateNewAllocation(Guid userId)
         {
-            var createNewAllocationCommand = new CreateNewAllocationByUserIdCommand(userId); 
-            await _commandHandler.Handle(createNewAllocationCommand);
+            var command = new CreateNewAllocationByUserIdCommand(userId); 
+            await _commandHandler.Handle(command);
         }
 
-        private async Task<User> GetUnallocatedUser(IEnumerable<User> users)
+        private static Guid GetUnallocatedUser(IEnumerable<Allocation> allocations)
         {
-            foreach (var user in users)
+            foreach (var allocation in allocations)
             {
-                var allocation = await GetAllocationByUserId(user.Id);
-
-                if (allocation.ExpiresAt == null)
-                {
-                    return user;
-                }
-
-                if (!allocation.Allocated || allocation.ExpiresAt < DateTime.UtcNow)
+                if (allocation.IsAllocated() == false)
```

Comment:
`!allocation.IsAllocated()` works the same with fewer characters :)

Reply:
nice, let me change it